### ISSUE TITLE
Reindex doesn't support multiple Types

### ISSUE THREAD

**Issue by dusda** [elastic/elasticsearch-net#1433]
So ElasticClient.ReIndex<T> is very handy for migrating indexes from one Type mapping to another, but what if you have more than one Type in your Index? ReIndex<T> doesn't let you use it on an existing Index and you can't specify more than one Type when using it, so I'm left with either writing my own implementation to reindex the thing, or re-work my model to move each Type mapping into it's own index.

**Comment by shanekenney** [comment]
@dusda NEST can reindex all types within an index. You should be able to use the following as an example (from ReindexTests):

```
Client.Reindex<object>(r => r
    .FromIndex("sourceindex")
    .ToIndex("targetindex")
```

**Comment by dusda** [comment]
That...works beautifully. Thank you.

What an odd syntax, though, wouldn't it make more sense to have method signature with no generic type, at least externally?

**Comment by gmarz** [comment]
Actually, it looks like we've uncovered a bug here.

Regardless of the type specified, whether some `T` or just `object`, reindex will still [search across all types](https://github.com/elastic/elasticsearch-net/blob/develop/src/Nest/Domain/Observers/ReindexObservable.cs#L58).

`Reindex()<T>` should respect the type when executing the search portion, and we should have a non-generic `Reindex()` as @dusda is suggesting, that defaults to all types, but allows for specifying multiple.

**Comment by Mpdreamz** [comment]
@gmarz @russcam vote to bump to 2.0.0 beta1? :+1: from me

**Comment by russcam** [comment]
@Mpdreamz Agreed :+1:

**Comment by russcam** [comment]
Closed via #1782
